### ISSUE TITLE
Can't stub public method inherited from a package visible class

### ISSUE THREAD

**Issue by gontard** [mockito/mockito#168]
I have these two classes

```java
class ParentTested {
    public Object getValue() {
        throw new NullPointerException();
    }
}
public class Tested extends ParentTested {}
```
i want to stub the `getValue`method in the following :

```java
@Test
public void aTest() throws Exception {
    Tested mock = mock(Tested.class);
    when(mock.getValue()).thenReturn(2);
    assertEquals(2, mock.getValue());
}
```
but it fails because in the line
```java
when(mock.getValue()).thenReturn(2);
```

the real method getValue is invoked. 

Am i doing something wrong ? or is it a Mockito bug/limitation ?

Notes:
 *  if the class `ParentTested` is public the test pass.
 * the method `getValue` is really public for a `Tested` since it is accessible outside the package.
 * [related question on stackoverflow](http://stackoverflow.com/questions/28442495/how-to-mock-riak-java-client)

**Comment by gontard** [comment]
If instead of throwing an Exception i return a value in `ParentTested`:

```java
class ParentTested {
    public Object getValue() {
        return 1;
    }
}
```
then i get a `MissingMethodInvocationException`:
```java
org.mockito.exceptions.misusing.MissingMethodInvocationException: 
when() requires an argument which has to be 'a method call on a mock'.
For example:
    when(mock.getArticles()).thenReturn(articles);

Also, this error might show up because:
1. you stub either of: final/private/equals()/hashCode() methods.
   Those methods *cannot* be stubbed/verified.
   Mocking methods declared on non-public parent classes is not supported.
2. inside when() you don't call method on mock but on some other object.

	at TestedTest.aTest(TestedTest.java:11)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)
```

**Comment by bric3** [comment]
Hi,

This is a duplicate of issue [212](https://code.google.com/p/mockito/issues/detail?id=212). This is a known limitation of CGLIB (in the version we are using) and in the way it generates proxies. I'm also caught in a similar situation with [Achilles](https://github.com/doanduyhai/Achilles).

* Unfortunately we are not going to patch CGLIB.
* Fortunately ByteBuddy mockmaker is around the corner, see #171

**Comment by gontard** [comment]
Hello,

thanks @bric3  for theses informations. 
So, I guess i should close this issue ?

**Comment by bric3** [comment]
Nope, let's continue the discussion there. GH is our new home and it's more comfortable than Google code ;)

**Comment by bric3** [comment]
should be fixed by #171
